### PR TITLE
Added efi disk

### PR DESCRIPTION
**Pull request by KnKay**
Hi all,

at the moment we try to get packer build proxmox with UEFI. 
As they use your api, it seems I need to add the efi disk here at well. 

I wonder, if adding simply the need here is fine. 

For what I understand the api a string will be just fine! 

Cheers! 

KnKay!

### CODE REVIEW

File: proxmox/config_qemu.go [hunk 1]
```diff
@@ -113,6 +114,10 @@ func (config ConfigQemu) CreateVm(vmr *VmRef, client *Client) (err error) {
 		params["bios"] = config.Bios
 	}
 
+	if config.EFIDisk != "" {
+		params["efidisk"] = config.EFIDisk
```
Comment (mleone87) [inline proxmox/config_qemu.go]: Thats should be efidisk0, did you check the api?

https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/qemu

**Review by mleone87 — COMMENTED** [review]
Hi

thats a complex parameter, it's like the other disk fields except that is separate for an efi disk, I think it needs a more robust logic

File: proxmox/config_qemu.go (hunk 1 above)
Comment (KnKay) [reply proxmox/config_qemu.go]: Guess you are right.
I will double this.

File: go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module github.com/Telmate/proxmox-api-go
+module github.com/KnKay/proxmox-api-go
```
Comment (pgporada) [inline go.mod]: I don't think you want to make this change.

Comment (KnKay) [reply go.mod]: You are right. Was for double checking a module relying on this. 
Thought the MR was rejected after the first coment (not the most experienced github user)

File: proxmox/config_qemu.go [hunk 3]
```diff
@@ -433,6 +438,10 @@ func NewConfigQemuFromApi(vmr *VmRef, client *Client) (config *ConfigQemu, err e
 	if _, isSet := vmConfig["bios"]; isSet {
 		bios = vmConfig["bios"].(string)
 	}
+	efidisk := "efidisk"
```
Comment (ggongaware) [inline proxmox/config_qemu.go]: "efidisk" as the default value ok?

**Comment by KnKay** [comment]
Hi all,

i did some more checks (of the api and the calls proxmox is doing). 
Proxmox is simply adding the _selectedstorage_:1 into the config if you use efidisk0. 

This seem to be not too tricky. Checked against 3 different clusters. 

As we needed it as well I also added trhe machine type!

**Comment by KnKay** [comment]
Hi all, is there any update on this? 
I opened a merge request for packer as well, whcih relies on this. :(

**Review by ggongaware — COMMENTED** [review]
> Hi all, is there any update on this?
> I opened a merge request for packer as well, whcih relies on this. :(


I had a question about this line>

efidisk := "efidisk"

Can you confirm the string "efidisk" as a default is valid, or should be empty string?

**Comment by KnKay** [comment]
You are right. Empty string would be better!

Grant Gongaware ***@***.***> schrieb am Mo., 21. Juni 2021,
18:07:

**Comment by g-m-o-n-e-y** [comment]
@ggongaware @mleone87 @pgporada We need this too. What's left to get this in?

btw: Thanks for working on this @KnKay!!

**Pull request merged**